crates/matrix-sdk/src/config/request.rs

```diff
@@ -44,18 +44,21 @@ pub struct RequestConfig {
     pub(crate) timeout: Duration,
     pub(crate) retry_limit: Option<u64>,
     pub(crate) retry_timeout: Option<Duration>,
+    pub(crate) max_concurrent_requests: usize,
```

Comment:
Instead of a `usize` that has a sentinel value (0), could we use `Option<NonZero...` here (everywhere in this PR)?

Reply:
ha. had that before but saw that the other values in the config are documented as "0 means the same" and adapted to that. Happy to change it to that.